src/analysis/interpolation/qgstininterpolator.cpp

```diff
@@ -145,14 +145,14 @@ void QgsTinInterpolator::initialize()
 
   if ( mInterpolation == CloughTocher )
   {
-    CloughTocherInterpolator *ctInterpolator = new CloughTocherInterpolator();
     NormVecDecorator *dec = dynamic_cast<NormVecDecorator *>( mTriangulation );
     if ( dec )
     {
+      auto ctInterpolator = std::make_unique<CloughTocherInterpolator>();
```

Comment:
where's the leak happening here? Is there exceptions raised somewhere in estimateFirstDerivatives or  setTriangulation ?

Reply:
`CloughTocherInterpolator *ctInterpolator = new CloughTocherInterpolator();` is allocated, but it used only in the `if (dec)` branch. So the simplest is to move this allocation in the `if (dec)` branch.

The `make_unique` is only to modernize this part and use smart pointer here.